Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -171,11 +160,53 @@ void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
 void UPSWorldSubsystem::Deinitialize()
 {
 	Super::Deinitialize();
-	PSHUDComponentInternal = nullptr;
-	PSCurrentSpotComponentInternal = nullptr;
-	SaveGameDataInternal = nullptr;
-	ProgressionSettingsDataInternal.Empty();
-	StarDynamicProgressMaterial = nullptr;
+}
+
+// Invoked after a game feature plugin is unloaded
+void UPSWorldSubsystem::OnGameFeatureUnloading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	const FString ProgressionPlugin = TEXT("ProgressionSystem");
```

Comment:
Please refactor the `ProgressionPlugin` definition as outlined below for improved efficiency and maintainability.

### What’s Wrong:
- **Redundant Declarations**: Defining `ProgressionPlugin` as a non-static variable in both functions leads to redundant memory allocation and makes maintenance harder if the module name changes.
- **Use of `FString` Instead of `FName`**: Defining the module name as an `FString` results in higher memory usage and slower comparisons than `FName`, which is better optimized for repeated static values.

### Expected Result:
```cpp
#include "CoreMinimal.h" // To include FName 
#include "Modules/ModuleInterface.h"

class PROGRESSIONSYSTEMRUNTIME_API FProgressionSystemRuntimeModule : public IModuleInterface
{
public:
	inline static const FName PSModuleName = TEXT("ProgressionSystem");

	//~IModuleInterface
	virtual void StartupModule() override;
	virtual void ShutdownModule() override;
	//~End of IModuleInterface
};

void UPSWorldSubsystem::OnGameFeatureUnloading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
{
	if (PluginURL.Contains(FProgressionSystemRuntimeModule::PSModuleName.ToString()))
	{
		//...
	} 
}

void UPSWorldSubsystem::OnGameFeatureLoading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
{
	if (PluginURL.Contains(FProgressionSystemRuntimeModule::PSModuleName.ToString()))
	{
		WolrdSubSystemInitialize();
	}
}
```

Reply:
![image](https://github.com/user-attachments/assets/a43fd74f-2823-4d41-88f0-1bb37d526fc7)
Don't see that code available on my side

Reply:
@h4rdmol You just show you do have the `FProgressionSystemRuntimeModule` file available, where new static property should be added. Then you use your static property everywhere instead of repeating the `ProgressionSystem' name

Reply:
Added :https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/6c826879c8535009fdebaac9917f52477ea72bf3